packages/main/src/plugin/container-registry.ts

```diff
@@ -2426,4 +2428,81 @@ export class ContainerProviderRegistry {
       });
     });
   }
+
+  async saveImages(options: ImagesSaveOptions): Promise<void> {
+    // group the images by engineId
+    const mapImages: Map<string, string[]> = options.images.reduce((map, img) => {
+      const imgIds = map.get(img.engineId) ?? [];
+      imgIds.push(img.id);
+      map.set(img.engineId, imgIds);
+      return map;
+    }, new Map<string, string[]>());
+
+    const isMultiProvider = mapImages.size > 1;
+    let errors = '';
+
+    for (const imageGroup of mapImages.entries()) {
+      const engine = this.internalProviders.get(imageGroup[0]);
+      if (!engine?.api) {
+        errors += `Unable to save images ${imageGroup[1].join(',')}. Error: No running provider for the matching images\n`;
+        continue;
+      }
+      if ('getImages' in engine.api) {
+        engine.api.getImages = function (
```

Comment:
here it looks like we'll override the function for each call to saveImages 

maybe move it to enhancePrototypeWithLibPod so it's only done once ?

Reply:
i extracted it to its own function. I can also move it to the contructor. Wouldn't it be better instead of updating enhancePrototypeWithLibPod as it's not related to libPod?

Reply:
ah nevermind, not possible as i don't have a ref of the various api there

Reply:
if i move it to libpod wouldn't it be restricted to podman providers? Don't we want it for docker as well?

Reply:
so as we're touching the prototype of Dockerode in fact it's modifying for all dockerode stuff (podman API or Docker API)

In fact Podman API is Dockerode + extra methods but it's in the same object

Reply:
Done 👍